src/types.cpp

```diff
@@ -73,14 +68,20 @@ size_t UnitsItem::index() const
 
 bool UnitsItem::isValid() const
 {
-    auto units = mPimpl->mUnits.lock();
-    if (units != nullptr) {
-        return mPimpl->mIndex < units->unitCount();
-    }
-
-    return false;
+    return mPimpl->mIndex < mPimpl->mUnits.lock()->unitCount();
```

Comment:
I think I could cause this to segfault.

Reply:
Not sure who changed the original code to this, but if I did then mea culpa. I agree that this is not good. I have therefore reverted to the original code (see commit 3d375612). However, I am now wondering whether we are still going to get 100% coverage... could it be the reason that someone/I switched to the code above?

Reply:
I can indeed confirm that we don't have 100% coverage anymore. Ok, need to find a way to "segfault" this now...

Reply:
Looking into it, `UnitsItem` is very much an internal type and, yet, it's in our API...

Reply:
Ok, maybe not since it's (only) needed by the `Annotator` class.

Reply:
Note that coverage has been addressed in commit 9b2745ae. However, this highlights another issue which is that the API allows us to create an `UnitsItem` while it clearly shouldn't. So... we should do what we have for the `Issue` class, i.e. only allow the creation of of `Issue` instances from our internal codebase. Am going to do that...

Reply:
First, commit 70796228 implements a better way of testing coverage for `UnitsItem::isValid()`.

Reply:
I guess that `UnitsItem` is similar to `VariablePair`, i.e. a convenience class. So, ok, I am leaving things as-is then, which means that I have reverted commit 7079622.